src/elements/Header/Header.js

```diff
@@ -52,10 +52,13 @@ function Header(props) {
     return <ElementType {...rest} className={classes}>{children}</ElementType>
   }
 
-  if ((image && typeof image !== 'boolean') || (icon && typeof icon !== 'boolean')) {
```

Comment:
These checks weren't entirely accurate. `image` or `icon` could have been passes as empty strings (valid itemShorthand) but still wouldn't be rendered. Also, the `!== boolean` is captured by the shorthand factory, which ignores booleans.